### DISCUSSION TITLE
Running pytest.main twice from the same python script

### DISCUSSION THREAD

**Discussion by nadavchernin** [pytest-dev/pytest#11536] [General]
Hi, 
I have script which run pytest.main twice.
There is a test which needs to be run in each session with different parametrized values
In parametrizing decorator i use function get_time:

import datetime
import pytest

def get_time():
    time_list = [datetime.datetime.now().second,
                 datetime.datetime.now().second + 1]
    return time_list

@pytest.mark.parametrize('time',get_time())
def test_1(time):
    print(time)

The main script as follows:

import pytest

sq = ["--collect-only",  # Here are pytest flags
      "rx_setup_tests/test_twice_parametrize.py::test_1",
      ]
pytest.main(sq)

sq = ["--collect-only",  # Here are pytest flags
      "rx_setup_tests/test_twice_parametrize.py::test_1",
      ]
pytest.main(sq)

The function get_time is called only once and the parameterized arguments stay the same in the second session.

Please your advice

Thank you very much, Nadav

**Comment by RonnyPfannschmidt** [comment]
This is in line with how python work's

Imports get cached

**Reply by nadavchernin** [reply]
Hi, Ronny
Not sure i understand you

**Reply by RonnyPfannschmidt** [reply]
The first time pytest runs, the tests get imported, and cached

The second time pytest Runs in the same process,those already imported modules are used again

**Reply by nicoddemus** [reply]
To complement:

> The second time pytest Runs in the same process,those already imported modules are used again

This is how Python works, not something specific to pytest.

In the example, `@pytest.mark.parametrize('time',get_time())` is executed at import time, so executed only once.
